### PR TITLE
Feature 295: project creation endpoint

### LINKED ISSUE #295: Create Project creation POST endpoint and DTO
For the next feature we wan't users to be able to create projects via the management page, for this we need an endpoint where the data of the new proposal is send to. In addition to this we need to update and extend our proposal DTO for the management panel. Updates to the DTO could also affect our current management page.

**AC:**

- [ ] Create an endpoint inside the management controller
- [ ] Update the proposal DTO
- [ ] add validation to the incoming values

### PR DESCRIPTION
**Pull request by KinimodEtlon**
closes #295

### CODE REVIEW

**Review by marhali — APPROVED** [review]

File: src/account/guards/project-creation.guard.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        const walletInfo = await this.walletInfoService.getCompleteInfo(user.wallet);
+        const createdProjects = await this.projectRepository.getAll({
+            find: {
+                'author.address': user.wallet.toLocaleLowerCase(),
+            },
+        });
+
+        if (walletInfo.balance < this.minOceanAmount) {
```
Comment (marhali) [inline src/account/guards/project-creation.guard.ts:30]: Should be balance <= 500$OCEAN

File: src/account/guards/project-creation.guard.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            throw new ForbiddenException(
+                `Users need at least ${this.minOceanAmount}OCEAN on the Mainnet, Polygon or BSC.`,
+            );
+        }
+
+        if (createdProjects.length > this.maxProjectAmount) {
```
Comment (marhali) [inline src/account/guards/project-creation.guard.ts:36]: Should be length >= maxProjectAmount

**Comment by marhali** [comment]
Categories: We should finally do the split between ProjectCategory and ProposalCategory. 
See https://github.com/ocean-tech-ship/ocean-pearl/tree/feature/768-project-creation for the frontend implementation

**Comment by KinimodEtlon** [comment]
> Categories: We should finally do the split between ProjectCategory and ProposalCategory. See https://github.com/ocean-tech-ship/ocean-pearl/tree/feature/768-project-creation for the frontend implementation

I agree, but i think that should be an issue for another ticket, this pr is big enough as it is even without this issue. If the frontend already covers the validation for this, we can update the backend some time else. I will create a ticket for this issue.

**Pull request merged**